src/elastic/src/client/requests/bulk/mod.rs

```diff
@@ -524,8 +524,11 @@ where
         let body = self.body.try_into_inner()?;
 
         match (self.index, self.ty) {
-            (Some(index), None) => Ok(BulkRequest::for_index(index, body)),
-            (Some(index), Some(ty)) => Ok(BulkRequest::for_index_ty(index, ty, body)),
+            (Some(index), ty) => match ty {
+                None => Ok(BulkRequest::for_index(index, body)),
+                Some(ref ty) if &ty[..] == "_doc" => Ok(BulkRequest::for_index(index, body)),
```

Comment:
We could actually use the `DEFAULT_DOC_TYPE` constant here. It's available under `elastic::types::document::DEFAULT_DOC_TYPE`.

I think after this I'll take the plunge proper and remove the multiple crates approach. Then we could just add an inherent method to `Type` that checks whether it's the default or not.